File: mobile/mobile/lib/models/today_record.dart
class TodayRecord {
  final String id;
  final String phoneNumber;
  final String userName;
  final String userType;
  final String interactionType;
  final String createdAt;

  TodayRecord({
    required this.id,
    required this.phoneNumber,
    required this.userName,
    required this.userType,
    required this.interactionType,
    required this.createdAt,
  });

  factory TodayRecord.fromJson(Map<String, dynamic> json) {
    return TodayRecord(
      id: json['id'] as String,
      phoneNumber: json['phoneNumber'] as String,
      userName: json['userName'] as String,
      userType: json['userType'] as String? ?? '일반',
      interactionType: json['interactionType'] as String? ?? 'UNKNOWN',
      createdAt: json['createdAt'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'phoneNumber': phoneNumber,
      'userName': userName,
      'userType': userType,
      'interactionType': interactionType,
      'createdAt': createdAt,
    };
  }
}

File: server/models/TodayRecord.js
const mongoose = require('mongoose');

const todayRecordSchema = new mongoose.Schema(
  {
    phoneNumber: {
      type: String,
      required: true,
    },
    userName: {
      type: String,
      required: true,
    },
    userType: {
      type: String,
      required: true,
    },
    interactionType: {
      type: String,
      required: true,
      enum: ['CALL', 'SMS']
    },
    createdAt: {
      type: Date,
      required: true,
      expires: 86400, // 24시간 후 자동 삭제
    },
  },
  {
    timestamps: false
  }
);

// phoneNumber와 userName에 대한 복합 인덱스 추가
todayRecordSchema.index({ phoneNumber: 1, userName: 1, interactionType: 1 }, { unique: true });
todayRecordSchema.index({ createdAt: -1 });

const TodayRecord = mongoose.model('TodayRecord', todayRecordSchema);

module.exports = TodayRecord; 